community/kernel/src/main/java/org/neo4j/kernel/impl/locking/LockClientStateHolder.java

```diff
@@ -66,22 +66,23 @@ public void stopClient()
 
     /**
      * Increment active number of clients that use current state instance.
-     * @return false if already stopped and not possible to increment active clients counter, true in case if counter
-     * was successfully incremented.
+     *
+     * @param client the locks client associated with this state; used only to create pretty exception
+     * with {@link LockClientStoppedException#LockClientStoppedException(Locks.Client)}.
+     * @throws LockClientStoppedException when stopped.
      */
-    public boolean incrementActiveClients()
+    public void incrementActiveClients( Locks.Client client )
     {
         int currentState;
         do
         {
             currentState = clientState.get();
             if ( isStopped( currentState ) )
             {
-                return false;
+                throw new LockClientStoppedException( client );
```

Comment:
So if we throw this exception then there's a chance user will see a `LockClientStoppedException` instead of `TransactionTerminatedException` as cause of tx failing, depending on which call happens to detect the fact that the tx has been terminated, will this be confusing to users?